### ISSUE TITLE
Support Visual studio 2019 or 2021

### ISSUE THREAD

**Issue by LADSoft** [LADSoft/OrangeC#435]
The microsoft compiler curently supported is VS2017.    There have been some issues with people needing to use VS2019, so, we either need to support it or wait for VS2021 to come out late next year and support that.

**Comment by GitMensch** [comment]
To just drop some breadcrumbs: VS 2019 Community Edition can currently be downloaded from https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=Community&rel=16  the build-tools only (which are used in CI)  can be downloaded currently from https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=BuildTools&rel=16#

Just rechecked: GnuCOBOL fails with different linker errors in VS2019 (works perfectly fine with GCC, clang and VisualStudio 2005-2017) - and as noted before it isn't the only project with issues like this.

**Comment by LADSoft** [comment]
thanks.

**Comment by bencz** [comment]
Hi David and @GitMensch 
Take a look on this: https://github.com/microsoft/STL

**Comment by chuggafan** [comment]
I think Microsoft's STL being released is a complete non-sequitor to this conversation about compiler errors and linker issues because of MSVC more conforming to the standard and the linker changing. 

As far as VS2019 stuff goes, at one point I was using the 2019 toolset and it was compiling fine but with issues post-link where it was giving off the wrong stuff entirely, but that might be due to undefined behavior we have, because it's everywhere like a pile of sphagetti most likely.

**Comment by LADSoft** [comment]
There are two issues apparently, one is the linker errors and the other is changes to the STL to make it conform better.

I fixed an issue the other day but when I downloaded VS2019 there were related issues, same problem different places.   All related to const-correctness of comparison functions passed to STL containers.   Thought I had checked that in last night but apparently missed it...

As far as the linker errors they seem gone at this point; I compiled everything with VS2019 then compiled everything with that version of the compiler and it seemed fine.
